std/experimental/ndslice/internal.d

```diff
@@ -1,11 +1,303 @@
 module std.experimental.ndslice.internal;
 
+import std.range.primitives;
 import std.traits;
-import std.meta; //: AliasSeq, anySatisfy, Filter, Reverse;
+import std.meta;
+import std.experimental.ndslice.slice;
+
+template PtrTuple(Names...)
+{
+    @LikePtr struct PtrTuple(Ptrs...)
+        if (allSatisfy!(isSlicePointer, Ptrs) && Ptrs.length == Names.length)
+    {
+        Ptrs ptrs;
+
+        void opOpAssign(string op)(sizediff_t shift)
+            if (op == `+` || op == `-`)
+        {
+            foreach (ref ptr; ptrs)
+                mixin (`ptr ` ~ op ~ `= shift;`);
+        }
+
+        auto opBinary(string op)(sizediff_t shift)
+            if (op == `+` || op == `-`)
+        {
+            auto ret = this;
```

Comment:
was `auto ret = this.ptrs;`.
This is fix for 16501. Unittest can be found below